source/jormungandr/jormungandr/external_services/external_service_provider_manager.py

```
@@ -0,0 +1,181 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# channel `#navitia` on riot https://riot.im/app/#/room/#navitia:matrix.org
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from __future__ import absolute_import, print_function, unicode_literals, division
+from importlib import import_module
+import logging
+import datetime
+from jormungandr import utils
+
+
+class ExternalServiceManager(object):
+    def __init__(
+        self, instance, external_service_configuration, external_service_getter=None, update_interval=600
+    ):
+        self.logger = logging.getLogger(__name__)
+        self._external_service_configuration = external_service_configuration
+        self._external_service_getter = external_service_getter
+        # List of external services grouped by navitia_service_type
+        self._external_services_legacy = {}
+        self._external_services_from_db = {}
+        self._external_services_last_update = {}
+        self._last_update = datetime.datetime(1970, 1, 1)
+        self._update_interval = update_interval
+        self.instance = instance
+
+    def init_external_services(self):
+        # Init external services from config file
+        for config in self._external_service_configuration:
+            # Set default arguments
+            if 'args' not in config:
+                config['args'] = {}
+            if 'service_url' not in config['args']:
+                config['args'].update({'service_url': None})
+            try:
+                service = utils.create_object(config)
+            except KeyError as e:
+                self.logger.error('Impossible to create external service with missing key: {}'.format(e.message))
+                continue
```

it would be better that you `raise` instead of `continue`